### DISCUSSION TITLE
Downloading a QFieldCloud project to Android - self hosted

### DISCUSSION THREAD

**Discussion by shapley** [opengisch/QField#5829] [Q&A]
Hi,

I've successfully uploaded a QGIS project to a self-hosted QC install. All has gone well and the project is showing on 'QFieldCloud admin' page and on an Android 13 device on the 'My Projects' tab. When I try to download the project it fails with the error (Downloading error.) with a 'red' dot prefixing the error.  No other explanation.

The Message Log shows: 

'Downloading project '7576e8f2-eaa2-4818-a643' finished with an error: Project 'myproject' file 'data.gpkg': Network error. Failed to download file 'data.gpkg'. System message:[HTTP/404] https://ourcloudserviceip/api/v1/packages/7476e8f2etc/late (see more in the QField error log).

The QField error log is empty.

I'm using QField 3.4.5 - Ebo (d260bb) on Android 13

I'm not sure whether this is an intended limitation to the QField app not being connected to 'app.qfield.com' by default or a mis-config of the docker image using the 'docker-compose.override.local.yml' instead of 'production' yml?. 

Any suggestions would be helpful.

**Comment by suricactus** [comment]
Hey, @shapley thanks for trying out QFieldCloud yourself.

> I'm not sure whether this is an intended limitation to the QField app not being connected to 'app.qfield.com' by default or a mis-config of the docker image using the 'docker-compose.override.local.yml' instead of 'production' yml?.

I can assure you there is no limitation on self-hosted vs app.qfield.cloud and the problems you are experiencing are outside the QField app.

> 'Downloading project '7576e8f2-eaa2-4818-a643' finished with an error: Project 'myproject' file 'data.gpkg': Network error. Failed to download file 'data.gpkg'. System message:[HTTP/404] [ourcloudserviceip/api/v1/packages/7476e8f2etc/late](https://ourcloudserviceip/api/v1/packages/7476e8f2etc/late) (see more in the QField error log).

If you get error HTTP 404, it seems a file with given name does not exist. Why I cannot say, needs deeper investigation. Seems there is a problem with the package, so check if jobs succeeded, the project table is updated with the latest package and the files are stored where they should be. I cannot guess much more, as there are quite a lot of combinations how you configured your system.

If you need dedicated support for your installation, please drop us a message sales@qfield.cloud .

**Comment by shapley** [comment]
Hi,

It seems the 'data.gpkg' , 'projectname_cloud_qfield.qgs' and 'projectname_cloud_qfield_attachments.zip' won't download as they were missing on upload to QFieldCloud. Will try and re-instate those files and report progress before closing call.

**Comment by shapley** [comment]
Hi,

It was in fact the 'data.gpkg' from the list of files uploaded (in the Qfield projects Overview List) that were missing but where are they uploaded to on a self-hosted server?

**Comment by shapley** [comment]
Hi,

Page not found (404)

![error_entry](https://github.com/user-attachments/assets/eedd62fb-6518-43a5-9d22-c0f463834b97)

May be the reason?

**Comment by shapley** [comment]
Hi,

Just as a reference I've got these errors when trying to download the qgis project from any Android device. Anyone with any idea please let me know. I'm guessing there is an issue with the '.env' config. As suggested by [suricactus] it could be anything.

Downloading project `fccdaecd-4ec5-4719-b234-6bce0b78d0a9` finished with an error: Project `projname`, file `data.gpkg`: Network error. Failed to download file `data.gpkg`. System message: [HTTP/0]
https://my-ip-address/api/v1/packages/fccdaecd-4ec5-4719-b234-6bce0b78d0a9/late (see more in the QField error log)

Downloading project `fccdaecd-4ec5-4719-b234-6bce0b78d0a9` finished with an error: Project `projname`, file `project_cloud_qfield_attachments.zip`: Network error. Failed to download file `projname_cloud_qfield_attachments.zip`. System message: [HTTP/0] https://my-ip-address/api/v1/packages/fccdaecd-4ec5-4719-b234-6bce0b78d0a9/late (see more in the QField error log)

Downloading project `fccdaecd-4ec5-4719-b234-6bce0b78d0a9` finished with an error: Project `projname`, file `projname_cloud_qfield.qgs`: Network error. Failed to download file `project_cloud_qfield.qgs`. System message: [HTTP/404] https://my-ip-address/api/v1/packages/fccdaecd-4ec5-4719-b234-6bce0b78d0a9/la (see more in the QField error log)
